### PR TITLE
NXP-24499, remove features-check contraints, adapt unit tests, cleanup

### PR DESCRIPTION
**Pull request by ThibArg**

### CODE REVIEW

**Comment by covolution** [comment]
:+1:

File: nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/TestPictureEventChain.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public void testPictureListener() throws IOException, OperationException {
         List<Tag> tags = tagService.getDocumentTags(session, picture.getId(), session.getPrincipal().getName());
 
         Assert.assertTrue(tags.size() > 0);
-        System.out.print(tags);
+        //System.out.print(tags);
```
Comment (driverpt) [inline nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/TestPictureEventChain.java]: Please remove all commented code

File: nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/TestVideoEventChain.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ public void testVideoChain() throws IOException, OperationException {
         List<Tag> tags = tagService.getDocumentTags(session, video.getId(), session.getPrincipal().getName());
 
         Assert.assertTrue(tags.size() > 0);
-        System.out.print(tags);
+        //System.out.print(tags);
```
Comment (driverpt) [inline nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/TestVideoEventChain.java]: Please remove all commented code

File: nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/TestVisionOp.java [hunk 3]
```diff
@@ -82,8 +85,9 @@ public void testOneBlobWithTags() throws IOException, OperationException {
         VisionResponse result = resultList.get(0);
         List<TextEntity> labels = result.getClassificationLabels();
         Assert.assertNotNull(labels);
-        Assert.assertTrue(labels.size() > 0);
-        System.out.print(labels);
+        assertEquals(MockVisionResponse.MOCK_RESULT_SIZE, labels.size());
+        assertEquals(MockVisionResponse.MOCK_TEXT, labels.get(0).getText());
+        //System.out.print(labels);
```
Comment (driverpt) [inline nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/TestVisionOp.java]: Same as above

File: nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/mock/MockVisionResponse.java [hunk 4]
```diff
@@ -28,14 +29,18 @@
...
     @Override
     public List<TextEntity> getClassificationLabels() {
-        return Arrays.asList(new TextEntity("label", 1.0f, "en"));
+        return Arrays.asList(new TextEntity(MOCK_TEXT, 1.0f, "en"));
     }
 
     @Override
     public List<TextEntity> getOcrText() {
-        return null;
+        return Arrays.asList(new TextEntity(MOCK_TEXT, 1.0f, "en"));
```
Comment (driverpt) [inline nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/mock/MockVisionResponse.java]: `Collections.singletonList`

File: nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java [hunk 5]
```diff
@@ -67,60 +68,60 @@ public void testProviderisLoaded() {
...
         Blob blob = new FileBlob(file);
         List<VisionResponse> results = getGoogleVisionProvider().execute(Arrays.asList(blob),
-                Arrays.asList(VisionFeature.TEXT_DETECTION), 5);
+                Arrays.asList(VisionFeature.TEXT_DETECTION.toString()), 5);
         assertEquals(1, results.size());
         List<TextEntity> texts = results.get(0).getOcrText();
         assertNotNull(texts);
         assertTrue(texts.size() > 0);
-        System.out.print(texts.get(0));
+        // System.out.print(texts.get(0));
```
Comment (driverpt) [inline nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java]: Commented code

File: nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java [hunk 6]
```diff
@@ -67,60 +68,60 @@ public void testProviderisLoaded() {
...
         Blob blob = new FileBlob(file);
         List<VisionResponse> results = getGoogleVisionProvider().execute(Arrays.asList(blob),
-                Arrays.asList(VisionFeature.IMAGE_PROPERTIES), 5);
+                Arrays.asList(VisionFeature.IMAGE_PROPERTIES.toString()), 5);
         assertEquals(1, results.size());
         List<ColorInfo> colors = results.get(0).getImageProperties().getColors();
         assertNotNull(colors);
         assertTrue(colors.size() > 0);
-        System.out.print(colors.get(0));
+        // System.out.print(colors.get(0));
```
Comment (driverpt) [inline nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java]: Commented code

File: nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java [hunk 7]
```diff
@@ -129,20 +130,21 @@ public void testMultipleFeatures() throws IOException, GeneralSecurityException
         List<TextEntity> texts = results.get(0).getOcrText();
         assertNotNull(texts);
         assertTrue(texts.size() > 0);
-        System.out.print(texts.get(0));
+        // System.out.print(texts.get(0));
```
Comment (driverpt) [inline nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java]: Commented code

File: nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java [hunk 8]
```diff
@@ -154,26 +156,45 @@ public void testAllFeatures() throws IOException, GeneralSecurityException {
         List<TextEntity> texts = result.getOcrText();
         assertNotNull(texts);
         assertTrue(texts.size() > 0);
-        System.out.print(texts.get(0));
+        // System.out.print(texts.get(0));
```
Comment (driverpt) [inline nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java]: Commented code

File: nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java [hunk 9]
```diff
@@ -154,26 +156,45 @@ public void testAllFeatures() throws IOException, GeneralSecurityException {
...
+    public void shouldFailWithWrongFeature() throws Exception {
+
+        File file = new File(getClass().getResource("/files/plane.jpg").getPath());
+        Blob blob = new FileBlob(file);
+        try {
+            @SuppressWarnings("unused")
+            List<VisionResponse> results = getGoogleVisionProvider().execute(Arrays.asList(blob),
+                    Arrays.asList(UUID.randomUUID().toString()), 5);
+            assertTrue("The call should have failed", false);
+        } catch (Exception e) {
```
Comment (driverpt) [inline nuxeo-vision-google/src/test/java/org/nuxeo/vision/google/test/TestGoogleVisionProvider.java]: Change exception variable name to `ignored`

File: nuxeo-vision-core/src/main/java/org/nuxeo/vision/core/service/Vision.java [hunk 10]
```diff
@@ -37,44 +37,44 @@
 
     /**
      * @param blob the image blob
-     * @param features the feature to request from the service
+     * @param feature(s) the feature to request from the service
```
Comment (kevinleturc) [inline nuxeo-vision-core/src/main/java/org/nuxeo/vision/core/service/Vision.java]: The param in javadoc should have the same naming than the parameter (for html links).

File: nuxeo-vision-core/src/main/java/org/nuxeo/vision/core/service/VisionProvider.java [hunk 11]
```diff
@@ -34,21 +35,16 @@
 
     /**
      * @param blobs the blobs to pass to the API
-     * @param features the feature to request from the provider
+     * @param feature(s) the feature to request from the provider
```
Comment (kevinleturc) [inline nuxeo-vision-core/src/main/java/org/nuxeo/vision/core/service/VisionProvider.java]: Same comment about javadoc link.

File: nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/mock/MockVisionProvider.java [hunk 12]
```diff
@@ -62,4 +64,21 @@ public boolean checkBlobs(List<Blob> blobs) throws IOException {
...
+    protected boolean featuresAreSupported(List<String> features) {
+
+        if(features == null) {
+            return false;
+        }
+
+        List<String> allowed = VisionFeature.asStringList();
+        for(String oneFeature : features) {
+            if(!allowed.contains(oneFeature)) {
+                return false;
```
Comment (kevinleturc) [inline nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/mock/MockVisionProvider.java]: ```return VisionFeature.asStringList().containsAll(features);``` should do the trick.

File: nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/mock/MockVisionResponse.java (hunk 4 above)
Comment (kevinleturc) [reply nuxeo-vision-core/src/test/java/org/nuxeo/vision/core/test/mock/MockVisionResponse.java]: This is not the only candidate.

**Comment by ThibArg** [comment]
Applied all the changes.
Even the _change the variable named involved in an ignore-warning to "ignored"_ ;-)

**Comment by driverpt** [comment]
👍

**Pull request closed without merging**